### DISCUSSION TITLE
Tools for measuring contributor growth and engagement on our repos

### DISCUSSION THREAD

**Discussion by chrisdavidmills** [mdn/mdn-community#515] [Content]
I was talking a little bit with @sideshowbarker about this on Matrix/Riot. While not the most urgent thing right now, it would be nice to think about a simple suite of tools that we can use for measuring contributor growth/engagement on our repos, including this one, browser-compat-data, etc.

From the conversation:

> For GitHub, anybody know of a tool that can show a graph of contributor growth for a particular repo?

> I mean, show how many new contributors were added each week (for this year, last year, etc.)

**Comment by peterbe** [comment]
I've never used it but Gitalytics had screenshots of really cool metrics graphs and stuff about activity in repos. 
But when I got to their site, it's all gone and replaced with this: https://github.blog/2020-05-14-reflecting-on-joining-github-and-whats-next-for-github-insights/
Good for them, I guess, but https://github.com/features/insights looks like it's just stuff that's part of the GitHub Enterprise suite which seems totally useless to us and extra useless to your point here in this issue. 

But it's worth pointing out that the API tools for getting to GitHub data is phenomenal. You can now so easily write your own little dashboards that use REST or GraphQL to analyze issues and commits and whatever. And with GitHub Actions, you can write bots that can update state based on cron jobs. So perhaps we can start with the most minimal requirement of what the measuring should be and work up from there.

**Comment by chrisdavidmills** [comment]
I'd love to chat about it at some point.

Also worth checking out, Dom from the W3C created this:

https://dontcallmedom.github.io/github-community-monitor/

Something similar for MDN repos would be great. I'll ask him if we can steal his code.

**Comment by peterbe** [comment]
Apart from this graph
<img width="1490" alt="Screen Shot 2020-11-19 at 10 21 28 AM" src="https://user-images.githubusercontent.com/26739/99685859-08020600-2a51-11eb-8c22-4cd427bd927a.png">
...I think that's a gorgeous dashboard. 

I love the idea that it sits outside the main thing. In this case, the dashboard lives under @_dontcallmedom so there's nothing stopping us or anybody doing the same.

**Comment by peterbe** [comment]
I built a dashboard last week using the `octokit` libraries (using the REST API) which displays a dashboard of issue assignment on the Yari repo. 
<img width="1446" alt="Screen Shot 2020-11-19 at 10 24 30 AM" src="https://user-images.githubusercontent.com/26739/99686379-88c10200-2a51-11eb-8145-1099a4a9d11a.png">
If I get around to it and finish it, I won't be pushing to have this included into Yari or Kuma or asking Ryan to review code on it. And that's kind of spirit that is so exciting about moving the content to GitHub that anybody can write high quality graphs and  dashboards without needing approval from the core team or having to resolve to web scraping.

**Comment by sideshowbarker** [comment]
https://wikimedia.biterg.io/app/kibana and https://insights.lfx.linuxfoundation.org/projects/ojsf%2Fojsf-jquery/dashboard are a couple more pretty nice dashboard-ish type things that show some of the same kinds of things it seems like we’d want too

**Comment by Ryuno-Ki** [comment]
If you're interested in time series charts, I find https://gource.io/ pretty beautiful.

**Comment by Rumyra** [comment]
@ddbeck possibly related to what we were discussing today?

**Comment by ddbeck** [comment]
@Rumyra Yeah, a little bit! I haven't got to the point where I've wanted to close track contributors (except for the number of total contributors, as we do the for [the BCD release notes stats section](https://github.com/mdn/browser-compat-data/releases/tag/v3.3.6)).

Another tool to make a note of though is GitHub' artifact exporter: https://github.blog/2021-05-18-github-artifact-exporter-open-source-release/
